dashboard/app/models/user.rb

```diff
@@ -830,6 +831,26 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_primary_authentication_option(email)
+    # If an auth option exists with same email, set it to the user's primary authentication option
+    existing_auth_option = authentication_options.find {|ao| ao.email == email}
```

Comment:
This logic doesn't handle students.  We'll also need to be able to accept only a hashed_email and find existing authentication options by matching hashed_email.

Reply:
thanks for catching that! i'll update logic and add a test case to cover this